### PR TITLE
[FLINK-14892][docs] Add documentation for checkpoint directory layout

### PR DESCRIPTION
**Pull request by klion26**
## What is the purpose of the change

In FLINK-8531, we change the checkpoint directory layout to
```
/user-defined-checkpoint-dir
    |
    + --shared/
    + --taskowned/
    + --chk-00001/
    + --chk-00002/
    + --chk-00003/
    ...
```
But the directory layout did not describe in the doc currently, and some users confused about this in the mail-list, this pr wants to add a description for the checkpoint directory layout in the documentation.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: ( no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by klion26** [comment]
cc @pnowojski 
cc @wuchong  for the translated Chinese version

File: docs/ops/state/checkpoints.md [hunk 1]
```diff
@@ -63,6 +63,24 @@ files. The meta data file and data files are stored in the directory that is
...
+    |
+    + --shared/
+    + --taskowned/
+    + --chk-00001/
+    + --chk-00002/
+    + --chk-00003/
+    ...
+{% endhighlight %}
+
+The **SHARED** directory is for state that is possibly part of multiple checkpoints, **TASKOWNED** is for state that must never by dropped by the JobManager, and **EXCLUSIVE** is for state that belongs to one checkpoint only.
```
Comment (klion26) [inline docs/ops/state/checkpoints.md]: The description order is according to the order of the above checkpoint directory layout.

File: docs/ops/state/checkpoints.md [hunk 2]
```diff
@@ -63,6 +63,24 @@ files. The meta data file and data files are stored in the directory that is
 configured via `state.checkpoints.dir` in the configuration files, 
 and also can be specified for per job in the code.
 
+The current checkpoint directory layout which introduced by FLINK-8531 is as follows:
```
Comment (pnowojski) [inline docs/ops/state/checkpoints.md]: Replace with
```
The current checkpoint directory layout ([introduced by FLINK-8531](https://issues.apache.org/jira/browse/FLINK-8531)) is as follows
```
?

**Comment by klion26** [comment]
@pnowojski thanks for the review, will address the comments.
As far as I know, `docs` do not need to manually update, it'll be done automatically.

**Comment by pnowojski** [comment]
Thanks for the update. (can be merged from my point of view already, but you can also ping me to merge this once the Chinese version is approved)

**Comment by klion26** [comment]
@pnowojski @wuchong thanks for the review and merging.

**Pull request merged**